go/carmen/proof.go

```diff
@@ -0,0 +1,56 @@
+package carmen
+
+import (
+	"github.com/Fantom-foundation/Carmen/go/common"
+	"github.com/Fantom-foundation/Carmen/go/common/witness"
+	"github.com/Fantom-foundation/Carmen/go/database/mpt"
+)
+
+// CreateWitnessProofFromNodes creates a witness proof from a list of strings.
+// Each string is an RLP node of the witness proof.
+func CreateWitnessProofFromNodes(elements ...string) WitnessProof {
+	proof := mpt.CreateWitnessProofFromNodes(elements)
+	return WitnessProof{proof}
+}
+
+type WitnessProof struct {
```

Comment:
To enable future refactoring hidden from external API users, we should hide this behind an interface.

Reply:
ok, done